pages/gallery/gallery.js

```diff
@@ -0,0 +1,420 @@
+        case 'java':
+          if (this.resolution === '16x') return `https://raw.githubusercontent.com/CompliBot/Default-Java/${this.version == 'latest' ? settings.versions[this.edition][0] : this.version}/${path.path}`
+          return `https://raw.githubusercontent.com/Compliance-Resource-Pack/Compliance-Java-${this.resolution}/Jappa-${this.version == 'latest' ? settings.versions[this.edition][0] : this.version}/${path.path}`
+      }
+    },
+    updateRoute(data, type) {
+      if (this.current[type] === data) return // avoid redundant redirection
+      this.current[type] = data
+
+      // user safe interaction
+      // check if resolution exist
+      if (!settings.resolutions.includes(this.current.resolution) && this.current.resolution !== '16x')
+        this.current.resolution = settings.resolution[0]
+
+      // check if edition exist
+      if (!settings.editions.includes(this.current.edition))
+
+        if (!settings.versions[this.current.edition.toLowerCase()].includes(this.current.version)) {
+          this.current.version = settings.versions[this.current.edition][0]
+          this.options.versions = settings.versions[this.current.edition]
+        }
+
+      this.$router.push({ params: { ...this.current } })
+    },
+    updateJSON(loadMessage = false) {
+      const params = [
+        { message: this.$root.lang().gallery.loading_message.tags, route: '/textures/types', key: 'tags' },
+        { message: this.$root.lang().gallery.loading_message.textures, route: `/gallery/textures/${this.edition}/${this.version}/${this.tag}/${this.search ? this.search : ''}`, key: 'textures' },
+        { message: this.$root.lang().gallery.loading_message.paths, route: `/gallery/paths/${this.edition}/${this.version}/${this.tag}/${this.search ? this.search : ''}`, key: 'paths' },
+        { message: this.$root.lang().gallery.loading_message.uses, route: `/gallery/uses/${this.edition}/${this.version}/${this.tag}/${this.search ? this.search : ''}`, key: 'uses' },
+        { message: this.$root.lang().gallery.loading_message.contributors, route: '/contributors/all', key: 'contributors' }
+      ]
+
+      if (this.current.resolution !== '16x') {
+        params.push(
+          { message: this.$root.lang().gallery.loading_message.contributions, route: '/contributions/all', key: 'contributions' },
+        )
+      }
+
+      this.fetchData(params, loadMessage)
+    },
+    fetchData(params, loadMessage) {
+      this.dataJSON = {}
+
+      this.loading = { status: false, comments: [], steps: params.length }
+      if (loadMessage) this.loading.status = true
+
+      params.forEach(param => {
+        axios.get(param.route)
+          .then(res => {
+            this.loading.comments.push(param.message)
+            this.dataJSON[param.key] = res.data
+          })
+          .catch(err => {
+            console.error(err) // loading will never ends, as we don't add "comment" into the comment list
```

Comment:
If data failed to be loaded, please add a div with a  stylized error as an output or just display a text saying "No <data> found."